tests/Spec/Nri/Ui/Util.elm

```diff
@@ -0,0 +1,33 @@
+module Spec.Nri.Ui.Util exposing (spec)
+
+import Expect
+import Nri.Ui.Util as Util
+import Test exposing (..)
+
+
+spec : Test
+spec =
+    describe "Nri.Ui.Util"
+        [ describe "safeIdString transforms strings as expected"
+            [ test "with a comma" <|
+                \() ->
+                    Util.safeIdString "Enable text-to-speech for Angela's account"
+                        |> Expect.equal "enable-text-to-speech-for-angela-s-account"
+            , test "removes leading non alpha characters" <|
+                \() ->
+                    Util.safeIdString "#@!!232now we are in business"
+                        |> Expect.equal "now-we-are-in-business"
+            , test "hard mode" <|
+                \() ->
+                    Util.safeIdString "!@#21something else interesting321$ ... hi"
+                        |> Expect.equal "something-else-interesting321-hi"
+            , test "with capitol letters" <|
+                \() ->
+                    Util.safeIdString "1232!@#%#@JFEKLfds-----SFJK3@#@jj23FDS........''''\"\"***"
+                        |> Expect.equal "jfeklfds-sfjk3-jj23fds-"
```

Comment:
I think I would expect to keep the starting numbers here? Just in case someone did something like `safeIdString "1-premade"`

Reply:
Yeah when I wrote the thing for Elm book club I was going with the rule most programming languages have for identifiers. In fact that follows the [HTML4 spec](https://www.w3.org/TR/html4/types.html#type-id) but not HTML 5 and up. MDN [recommends](https://developer.mozilla.org/en-US/docs/Web/HTML/Global_attributes/id#sect2) starting with a letter, but you're right that this has a pretty obvious path to creating duplicate IDs which are an actual problem, so I'll change that.

Reply:
ehhhh never mind. It does not work. Javascript cannot handle it in `querySelector` or `querySelectorAll`, CSS cannot handle it. It only works with `getElementById`. https://codepen.io/ap-nri/pen/OJENQLY

Reply:
Huh! I had no idea that numbers at the start were no good!